client/ayon_deadline/repository/custom/plugins/GlobalJobPreLoad.py

```diff
@@ -651,6 +604,141 @@ def inject_ayon_environment(deadlinePlugin):
         raise
 
 
+def _wait_for_in_progress(job, export_path):
+    """Check if another worker started extractenvironments already.
+
+    extractenvironments might be expensive operation, so first worker who
+    starts doing it creates empty file before it. All other workers should
+    wait until this file gets renamed to final name.
+    Raises:
+        (RuntimeError) if extraction takes more
+            than EXTRACT_ENVIRONMENT_TIMEOUT seconds
+    """
+    export_in_progress_path = f"{export_path}.tmp"
+    timeout = int(
+        job.GetJobEnvironmentKeyValue("AYON_EXTRACT_ENVIRONMENT_TIMEOUT")
+        or EXTRACT_ENVIRONMENT_TIMEOUT
+    )
+    while os.path.exists(export_in_progress_path):
+        file_modified  = datetime.fromtimestamp(
+            os.path.getmtime(export_in_progress_path)
+        )
+        date_diff = datetime.now() - file_modified
+        if date_diff > timedelta(seconds=timeout):
+            print(
+                "Previous extract environment process stuck for "
+                f"'{timeout}' sec. Starting it from scratch."
+            )
+            try:
+                os.remove(export_in_progress_path)
+            except (OSError, PermissionError):
+                raise RuntimeError(
+                    f"Failed to remove progress file '{export_in_progress_path}'."
+                )
+        print("Extract environment process already triggered, waiting")
+        sleep(2)
+
+
+def _get_output_dir(job):
+    """Look for output dir where metadata.json should be created also."""
+    output_urls = ["OutputFilePath", "Output", "SceneFile"]
+    for output in output_urls:
+        output_value = job.GetJobPluginInfoKeyValue(output)
+        if not output_value:
+            continue
+        _, ext = os.path.splitext(output_value)
+        if ext:
+            return os.path.dirname(output_value)
+        else:
+            return output_value
```

Comment:
```suggestion
        if os.path.isdir(output_value):
            return output_value
        return os.path.dirname(output_value)
```

Reply:
`isdir` checks if that folder exists, it might not yet.